### PR TITLE
Guard against duplicate key creation

### PR DESCRIPTION
**Pull request by steveoh**
refs #286

### CODE REVIEW

File: src/developer/functions/https/createKey.js [hunk 1]
```diff
@@ -149,3 +157,43 @@ export const generateRegexFromPattern = (inputPattern) => {
...
+    // if the modes are different they are not the same key
+    if (key?.flags?.production !== (data.mode === 'production')) {
+      continue;
+    }
+
+    // if the key's pattern or ip matches
+    if (
+      (key?.flags?.server === false && key?.pattern === data.pattern) ||
+      (key?.flags?.server && key?.pattern === data.ip)
+    ) {
```
Comment (stdavis) [inline src/developer/functions/https/createKey.js:191]:
```suggestion
    if (key?.pattern === (key?.flags?.server) ? data.ip : data.pattern) {
```

Slightly more simple way to do it...

File: src/developer/functions/https/keys.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ try {
  *   notes: string
  * }[]>} an array of keys run through the minimalKeyConversion converter
  */
-export const getKeys = async (uid) => {
+export const getKeys = async (uid, skipAnalytics = false) => {
```
Comment (stdavis) [inline src/developer/functions/https/keys.js:27]: You need to update the doc string for the newly added prop.

File: src/developer/functions/https/keys.js [hunk 3]
```diff
@@ -39,25 +39,31 @@ export const getKeys = async (uid) => {
...
-    results.forEach((result, index) => {
-      if (index < keys.length) {
-        keys[index].usage = result ?? 0;
-      } else {
-        keys[index - keys.length].lastUsed = result ?? -1;
-      }
-    });
-  } catch (ex) {
-    error('redis error', ex);
+  if (skipAnalytics) {
```
Comment (stdavis) [inline src/developer/functions/https/keys.js:42]:
```suggestion
  if (!skipAnalytics) {
```
Should this be flipped?

Comment (steveoh) [reply src/developer/functions/https/keys.js:42]: no, but the variable should be renamed...

**Review by stdavis — APPROVED** [review]

**Pull request merged**